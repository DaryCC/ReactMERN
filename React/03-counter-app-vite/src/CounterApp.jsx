import React,{useState} from 'react';
// import ReactDOM from 'react-dom/client';
// import {fragment} from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value})=>{
  // console.log(value);
  const [counter, setCounter] = useState(value);

  const incrementaContador=()=> {
    return setCounter(()=> counter+1);
  }
  const decrementarContador =()=> {
    setCounter(counter-1);
  }
  const resetCounter = ()=>{
    setCounter(value);
  };

  return (
    <>
   <h1>CounterApp</h1>
   <h2>{counter}</h2>
      <button onClick={incrementaContador} name=''>+1</button>
      <button onClick={decrementarContador}>-1</button>
      <button onClick={resetCounter}>Reset</button>
    </>
  );
};

CounterApp.propType = {
  value: PropTypes.number.isRequired,
}

CounterApp.defaultProps={
  value:10,
};
