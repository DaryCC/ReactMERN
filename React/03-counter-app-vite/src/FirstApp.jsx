import {Fragment} from 'react';
import PropTypes from 'prop-types';

const mensaje='Hola Nadia.';

const newMessageFunction =()=>{
  return 'Amo a los gatitos.';
};

export const FirstApp = ({name,title,subtitle}) => {
  // console.log(props);
  if (!title){
    throw new error ('El title no existe');
  }

  return (
    <>
      <h1>{title} </h1>
      <p>Propiedad subtítulo: {subtitle}</p>
      <ul >
        <li>{newMessageFunction()}</li>
        <li>{mensaje}</li>
        <li>{name}</li>
        <li>La resta es {12-12}</li>
        <li>Negativo {12-15}</li>
      </ul>
    </>
  )
}


FirstApp.propTypes = {
  title:PropTypes.string.isRequired,
  subtitle:PropTypes.number,

};

FirstApp.defaultProps = {
  // title: 'Títuto default',
  subtitle: 666,
  name: 'Nadia Barriga',
}
