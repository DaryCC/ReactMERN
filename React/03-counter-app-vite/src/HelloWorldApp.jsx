
// export function App() {
//   //document.createElement...
//   return (<h1>Hola mundo</h1>);
// }
import React from 'react'

const HelloWorldApp = () => {
  return (
    <h1>
      Hola Mundo Cruel
    </h1>
  )
}

export default HelloWorldApp;


export const Dary = () => {
  return (
    <h2>
      Daryberto Canseco Cort√©s.
    </h2>
  )
}

