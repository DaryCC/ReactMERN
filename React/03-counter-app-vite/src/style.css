html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    /* border: yellow dotted; */
    padding: 70px;
    /* padding-bottom:150px;
       padding-top:150px; */
}
button{
    padding: 15px;
    background-color:grey;
    margin-right:1.5em;
}
