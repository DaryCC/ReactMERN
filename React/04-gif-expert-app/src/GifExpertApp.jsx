import React, { useState } from 'react'
import {AddCategory} from './components/AddCategory';


export const GifExpertApp = () => {
  const [categories, setcategories] = useState(['One Punch', 'Dragon']);
  // console.log(categories);
const onAddCategory = () =>{
  setcategories([...categories, 'Hola Mundo']);
}

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory setcategories={ setcategories}/>
      {/* <button onClick={onAddCategory}>GifExpertApp</button> */}
      <ol>
        {
          categories.map( category => {
            return <li key={ category}>{category}</li>
          })
        }
      </ol>
    </>
  )
}
