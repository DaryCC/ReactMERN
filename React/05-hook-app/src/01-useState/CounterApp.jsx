import React,{useState} from 'react'

const CounterApp = () => {

  // const [{counter1,counter2,counter3}, setCounter] = useState({
  const [state, setCounter] = useState({
    counter1:0,
    counter2:5,
    counter3:10
  });
  const {counter1,counter2,counter3}=state;
  return (
    <>
      <h1>Counter1:{counter1}</h1>
      <h1>Counter2:{counter2}</h1>
      <h1>Counter3:{counter3}</h1>
      <hr/>
      <button onClick={()=>setCounter({
        ...state,
        counter3:counter3 +10,
      })}>+1</button>

    </>
  )
}

export default CounterApp
