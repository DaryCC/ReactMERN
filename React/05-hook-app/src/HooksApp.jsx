import React from 'react'

const HooksApp = (props) => {
  return (
    <h1>
      HooksApp
    </h1>
  )
}

export default HooksApp
