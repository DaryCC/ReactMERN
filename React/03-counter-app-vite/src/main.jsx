import React from 'react';
import ReactDOM from 'react-dom/client';
// import PropTypes from 'prop-types';
// import Hello,{Dary} from './HelloWorldApp';
// import {FirstApp  } from './FirstApp';
import {CounterApp} from './CounterApp';

import './style.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <FirstApp /> */}
    {/* <FirstApp title="Nadia" subtitle={1+2+3}/> */}
    <CounterApp value={12}/>
  </React.StrictMode>
);
