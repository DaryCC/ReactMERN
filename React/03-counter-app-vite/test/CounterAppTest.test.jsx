import {getByTestId, render,screen} from '@testing-library/react';
import {CounterApp} from '../src/CounterApp';
import { Fragment } from 'react';

const valor=100;

describe('Pruebas en <CounterApp />' ,()=>{
  test('debe ser correcto si el snapshot coincide', () => {
    const {container, getByText}=render(<CounterApp value={valor}/>);
    expect(container).toMatchSnapshot();


  });

  test('debe mostrar el valor inicial 100', () => {

    const { container, getByText } = render(<CounterApp value={valor} />);
    const h2= container.querySelector('h2');
    expect(h2.innerHTML).toContain('100');
  });
});
