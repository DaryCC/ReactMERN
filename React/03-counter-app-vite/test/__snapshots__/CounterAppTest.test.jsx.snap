// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe ser correcto si el snapshot coincide 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    name=""
  >
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
